bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/protocols/ScalarWebAppControlProtocol.java

```diff
@@ -0,0 +1,512 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The implementation of the protocol handles the AppControl service.
+ * 
+ * Note: haven't figured out text encryption yet - so I have the basics commented out for future use
+ *
+ * @author Tim Roberts - Initial contribution
+ * @param <T> the generic type for the callback
+ */
+@NonNullByDefault
+class ScalarWebAppControlProtocol<T extends ThingCallback<String>> extends AbstractScalarWebProtocol<T> {
+    /** The logger */
+    private final Logger logger = LoggerFactory.getLogger(ScalarWebAppControlProtocol.class);
+
+    // Varous channel constants
+    private static final String APPTITLE = "apptitle";
+    private static final String APPICON = "appicon";
+    private static final String APPDATA = "appdata";
+    private static final String APPSTATUS = "appstatus";
+    private static final String TEXTFORM = "textform";
+    private static final String STATUS = "status";
+    private static final String START = "start";
+    private static final String STOP = "stop";
+
+    // The intervals used for refresh
+    private static final int APPLISTINTERVAL = 60000;
+    private static final int ACTIVEAPPINTERVAL = 10000;
+
+    /** The encryption key to use */
+    // private final @Nullable String pubKey;
+
+    /** The lock used to modify app information */
+    private final Lock appListLock = new ReentrantLock();
+
+    /** The last time the app list was accessed */
+    private long appListLastTime = 0;
+
+    /** The applications */
+    private final List<ApplicationList> apps = new CopyOnWriteArrayList<ApplicationList>();
+
+    /** The lock used to access activeApp */
+    private final Lock webAppStatusLock = new ReentrantLock();
+
+    /** The last time the activeApp was accessed */
+    private long webAppStatusLastTime = 0;
+
+    /** The active app. */
+    private @Nullable WebAppStatus webAppStatus = null;
+
+    /**
+     * Instantiates a new scalar web app control protocol.
+     *
+     * @param context the non-null context to use
+     * @param context the non-null context to use
+     * @param service the non-null service to use
+     * @param callback the non-null callback to use
+     */
+    ScalarWebAppControlProtocol(final ScalarWebProtocolFactory<T> factory, final ScalarWebContext context,
+            final ScalarWebService service, final T callback) {
+        super(factory, context, service, callback);
+
+        // String localPubKey = null;
+        // final ScalarWebService enc = getService(ScalarWebService.ENCRYPTION);
+        // if (enc != null) {
+        //     try {
+        //         final PublicKey publicKey = enc.execute(ScalarWebMethod.GETPUBLICKEY).as(PublicKey.class);
+        //         localPubKey = publicKey.getPublicKey();
+        //     } catch (final IOException e) {
+        //         logger.debug("Exception getting public key: {}", e.getMessage());
+        //     }
+        // }
+
+        // pubKey = localPubKey;
```

Comment:
dead code?

Reply:
no - still unknown - keeping the code until I figure out how to do the encryption correctly